packages/components/src/components/SearchBarV2.tsx

```diff
@@ -0,0 +1,185 @@
+      renderNoResults({ Fragment, state: { query, status } }, root) {
+        if (!query || status === 'loading') {
+          render(<Fragment />, root);
+        } else {
+          render(
+            <Fragment>
+              <NoResultsContainer>No results for "{query}"</NoResultsContainer>
+            </Fragment>,
+            root
+          );
+        }
+      },
+      render({ children, state, Fragment, components }, root) {
+        render(
+          <Fragment>
+            <Container>
+              <ResultsContainer>{children}</ResultsContainer>
+              <SidePreview
+                accentColor={accentColor}
+                item={state.context.preview}
+                components={components}
+              />
+            </Container>
+            <SearchBy
+              href="https://algolia.com"
+              target="_blank"
+              rel="noreferrer"
+            >
+              Search by&nbsp;
+              <AlgoliaLogo />
+            </SearchBy>
+          </Fragment>,
+          root
+        );
+      },
+      openOnFocus: true,
+      getSources({ query }) {
+        if (!query) {
+          return [];
+        }
+        return debounced([
+          {
+            sourceId: 'algoliaIndex',
+            getItems: () =>
+              getAlgoliaResults({
+                searchClient,
+                queries: [
+                  {
+                    indexName,
+                    query,
+                    params: {
+                      hitsPerPage: 20,
+                    },
+                  },
+                ],
+              }),
+            onActive({ item, setContext }) {
+              setContext({ preview: item });
+            },
+            getItemUrl: ({ item }) => item.url,
+            templates,
+          },
+        ]);
+      },
+      reshape({ sourcesBySourceId, sources }) {
+        if (sources.length === 0) {
+          return sources;
+        }
+
+        const { algoliaIndex, ...restSources } = sourcesBySourceId;
+
+        const items = algoliaIndex.getItems();
+        const itemsSources = items
+          .slice()
+          // put current website's section on top
+          .sort((a) => (a.domain.startsWith(window.location.origin) ? 1 : -1))
+          .map((item) => item.source);
+
+        const sourcesPerSite = itemsSources.reduce<Record<string, any>>(
+          (acc, sourceId) => {
+            if (!acc[sourceId]) {
+              acc[sourceId] = {
+                ...algoliaIndex,
+                sourceId,
+                getItems: () =>
+                  items.filter((item) => item.source === sourceId),
+                templates,
+              };
+            }
+            return acc;
+          },
+          {}
+        );
+
+        return Object.values({ ...sourcesPerSite, ...restSources });
+      },
+    });
+    search.current = s;
+
+    return () => {
+      s.destroy();
+    };
+  }, [accentColor, placeholder]);
+
+  // listen for CTRL+K
```

Comment:
I fixed it, now cmd+k not only open modal but closes also

Reply:
Thanks @B2o5T !!